dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/EventImporterValidationRunner.java

```diff
@@ -27,54 +27,63 @@
+import static org.hisp.dhis.dxf2.importsummary.ImportStatus.WARNING;
 
+import java.util.ArrayList;
 import java.util.List;
 
-import org.apache.commons.logging.Log;
+import lombok.RequiredArgsConstructor;
+
 import org.hisp.dhis.dxf2.events.event.Event;
 import org.hisp.dhis.dxf2.events.importer.context.WorkContext;
+import org.hisp.dhis.dxf2.events.importer.shared.ImmutableEvent;
+import org.hisp.dhis.dxf2.importsummary.ImportSummary;
+import org.springframework.stereotype.Component;
 
-/**
- * Simple interface that provides processing capabilities on events.
- *
- * @author maikel arabori
- */
-public interface EventProcessing
+@Component
+public class EventImporterValidationRunner
 {
-    void process( WorkContext workContext, List<Event> events );
-
-    Log log = getLog( EventProcessing.class );
-
-    class ProcessorRunner
+    /**
+     * Validates the events using the supplied list of validators.
+     * <p>
+     * Only returns the ImportSummary for Events that *did* not pass validation
+     *
+     * @param payload Validation Runner payload to run against
+     * @return returns the ImportSummary for Events that did not pass validation
+     */
+    public List<ImportSummary> run( ValidationRunnerPayload payload )
     {
-        private final WorkContext workContext;
-
-        private final List<Event> events;
-
-        public ProcessorRunner( WorkContext workContext, List<Event> events )
-        {
-            this.workContext = workContext;
-            this.events = events;
-        }
+        final List<ImportSummary> importSummaries = new ArrayList<>( 0 );
 
-        public void run( final List<Class<? extends Processor>> processors )
+        for ( final Event event : payload.events )
         {
-            for ( final Event event : events )
+            for ( Checker validationCheck : payload.validators )
             {
-                for ( Class<? extends Processor> processor : processors )
+                final ImportSummary importSummary = validationCheck.check( new ImmutableEvent( event ),
+                    payload.workContext );
+
+                if ( importSummary.isStatus( ERROR ) || importSummary.isStatus( WARNING ) )
                 {
-                    try
-                    {
-                        final Processor pre = processor.newInstance();
-                        pre.process( event, workContext );
-                    }
-                    catch ( InstantiationException | IllegalAccessException e )
+                    importSummaries.add( importSummary );
+                    if ( importSummary.isStatus( ERROR ) )
                     {
-                        log.error( "An error occurred during Event import processing", e );
+                        break;
                     }
                 }
             }
         }
+
+        return importSummaries;
+    }
+
+    @RequiredArgsConstructor( staticName = "of" )
+    static class ValidationRunnerPayload
```

Comment:
I think we mostly call these `...Parameters` or `...Params`. I guess it does not hurt to make it public to it can be called from any package or is this intentionally package local?

Reply:
see @teleivo 's comment above, I removed this class